### PR TITLE
Java team's branch merge to main

### PR DESCRIPTION
**Pull request by pillai-karthik**

### CODE REVIEW

File: src/components/list/RenderTableList.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import { h } from "gridjs";
+import { Grid } from "gridjs-react";
+import React from "react";
+import apiFactory from "../../utils/api";
+interface IProps {
```
Comment (paras-1999) [inline src/components/list/RenderTableList.tsx:5]: don't write interface here inside type> index and then import

File: src/AppRoutes.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { Routes, Route, Navigate } from "react-router-dom";
 import { AdminGuard, TenantGuard, UserGuard } from "./components/auth-gaurd";
 import { AuthGuard, access } from "./components/gateway/auth-guard";
 import Spinner from "./components/loader/Loader";
+import GetTables from "./pages/features/saas/get-tables/GetTables";
```
Comment (paras-1999) [inline src/AppRoutes.tsx:6]: do lazy loading over here

File: src/pages/features/saas/manage-table/ManageTable.tsx [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+    const pageParameters: IPagination = {
+      pageNumber: (currentPage + 1).toString(),
+      pageSize: "6",
+    };
+
+    dispatch(getAllTables(pageParameters));
+  };
+  return (
+    <div className="createbody">
```
Comment (paras-1999) [inline src/pages/features/saas/manage-table/ManageTable.tsx:114]: use conditional rendering for the loading and error page

File: src/pages/features/saas/search-data/SearchData.tsx [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+        }
+      });
+      if (check === 1) {
+        results.push(val);
+      }
+    });
+    setFilterData(results);
+  };
+  return (
+    <div>
```
Comment (paras-1999) [inline src/pages/features/saas/search-data/SearchData.tsx:171]: do conditional rendering

**Review by paras-1999 — CHANGES_REQUESTED** [review]
look for the inline comments that i add regarding the conditional rendering and interface

File: src/pages/features/saas/search-data/SearchData.tsx [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+  const dispatch = useAppDispatch();
+  const searchData = useAppSelector(
+    (state) => state.searchDataWithQueryFieldState
+  );
+  const tableData = useAppSelector((state) => state.getTableState);
+  const tableColName = useAppSelector((state) => state.getTableSchemaState);
+  const tenantDetails = useAppSelector((state) => state.getTenantDetailState);
+  const [checkDisable, setCheckDisable] = useState(false);
+
+  const [tenantId, setTenantId] = useState("");
```
Comment (mihir-neosoft) [inline src/pages/features/saas/search-data/SearchData.tsx:37]: combine states in possible

**Review by mihir-neosoft — CHANGES_REQUESTED** [review]
.vscode files are unnecessary.
Try to avoid using any as type.
Take pull from main regularly.

File: src/pages/features/saas/add-table/CreateTable.tsx [hunk 6]
```diff
@@ -0,0 +1,973 @@
...
+    setModalState("modal-two");
+  };
+
+  const handleShowModalTwoclose = () => setModalState("close");
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  const toUppercase = (str: string) => {
```
Comment (mihir-neosoft) [inline src/pages/features/saas/add-table/CreateTable.tsx:59]: Please make a util file and add this tyeof function in that

File: src/pages/features/saas/search-data/SearchData.tsx (hunk 4 above)
Comment (ojahagirdar) [reply src/pages/features/saas/search-data/SearchData.tsx:212]: resolved

**Review by mangeshdase — APPROVED** [review]

**Review by santoshshinde2012 — APPROVED** [review]

**Pull request merged**